src/Mod/Fem/Gui/ViewProviderFemPostObject.cpp

```diff
@@ -618,12 +618,24 @@ void ViewProviderFemPostObject::WritePointData(vtkPoints* points,
     }
 }
 
-void ViewProviderFemPostObject::setRangeOfColorBar(double min, double max)
+void ViewProviderFemPostObject::setRangeOfColorBar(float min, float max)
 {
     try {
         if (min >= max) {
-            min = max - 10 * std::numeric_limits<double>::epsilon();
-            max = max + 10 * std::numeric_limits<double>::epsilon();
+            float eps = std::numeric_limits<float>::epsilon();
```

Comment:
Can you add a comment explaining what this code is trying to accomplish? It seems like a lot of complexity for what (in the code) looks like an input error.

Reply:
Comment added.
Basically, the code handles the case of min value equal to max value, and also the case where min is expected to be equal to max but turns out to be greater due to floating point error.